### PR TITLE
Migrate RecoParticleFlow/PFTracking to esConsumes

### PR DESCRIPTION
**Pull request by marksan87**
#### PR description:

Migration of `RecoParticleFlow/PFTracking` package to esConsumes to address #31061

#### PR validation:

Run standard limited matrix evaluation

`runTheMatrix.py -l limited -j 8 --ibeos`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport

@hatakeyamak

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by hatakeyamak** [comment]
@bendavid @laurenhay @rappoccio

**Comment by jpata** [comment]
@cmsbuild please test

I can't find the previous test in jenkins, seems stuck/lost.

File: RecoParticleFlow/PFTracking/plugins/HGCalTrackCollectionProducer.cc [hunk 1]
```diff
@@ -81,12 +85,17 @@ HGCalTrackCollectionProducer::HGCalTrackCollectionProducer(const edm::ParameterS
...
-      useIterTracking_(iConfig.getParameter<bool>("useIterativeTracking")) {
+      useIterTracking_(iConfig.getParameter<bool>("useIterativeTracking")),
+      magneticFieldToken_(esConsumes<edm::Transition::BeginLuminosityBlock>()),
+      tkerGeomToken_(esConsumes<edm::Transition::BeginLuminosityBlock>()) {
   LogDebug("HGCalTrackCollectionProducer")
       << " HGCalTrackCollectionProducer::HGCalTrackCollectionProducer " << std::endl;
 
   const edm::ParameterSet& geoconf = iConfig.getParameterSet("hgcalGeometryNames");
   hgc_names_[0] = geoconf.getParameter<std::string>("HGC_ECAL");
+  for (unsigned i = 0; i < hgcGeometryTokens_.size(); i++) {
```
Comment (jpata) [inline RecoParticleFlow/PFTracking/plugins/HGCalTrackCollectionProducer.cc:96]:
```suggestion
  for (const auto i : hgcGeometryTokens_) {
```

Comment (jpata) [reply RecoParticleFlow/PFTracking/plugins/HGCalTrackCollectionProducer.cc:96]: In fact, I'm a little confused why you need an `std::array` of length 1 here.

**Comment by jpata** [comment]
No unexpected reco changes detected. Code changes look good, a minor comment inline.

Looks like there's another `get` to migrate in `GoodSeedProducer`. You can get this info from the bot Comparison Summary. Summary -> CMS clang-tidy checks EventSetupRecord::get warnings
```
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_3_X_2021-04-08-2300/src/RecoParticleFlow/PFTracking/plugins/GoodSeedProducer.cc:620:3: warning: direct call of function EventSetupRecord::get(ESHandle&) is deprecated and should be replaced with a call to EventSetup::getHandle(ESGetToken&). To use ESGetToken see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ED_module To get data with the token see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_EventSetup_wit [cms-esrget]
```

File: RecoParticleFlow/PFTracking/plugins/HGCalTrackCollectionProducer.cc (hunk 1 above)
Comment (marksan87) [reply RecoParticleFlow/PFTracking/plugins/HGCalTrackCollectionProducer.cc:90]: This format was kept to allow for future extrapolation to HGCal hadronic sections.

**Comment by jpata** [comment]
@cmsbuild please test

**Comment by jpata** [comment]
+reconstruction
- technical esConsumes migration, no observed/expected reco changes (140.53 seems unrelated)

Thanks for this!

**Comment by silviodonato** [comment]
+1

> The workflows 140.53 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

**Pull request merged**